src/client.js

```diff
@@ -123,7 +130,26 @@ export default class Client {
 
     this.on('no_longer_subscribed', (event) => {
       this._ping.stop();
-      this.reconnect();
+
+      if (this.hardReconnectCount >= HARD_RECONNECT_THRESHOLD) {
+        this.logger.error(`no_longer_subscribed has been called ${this.hardReconnectCount} times and the threshold is ${HARD_RECONNECT_THRESHOLD}, not attempting to reconnect`);
+        clearInterval(this.leakyReconnectTimer);
+        return;
+      }
+
+      this.hardReconnectCount++;
+
+      this.logger.info('no_longer_subscribed received, attempting hard reconnect');
+
+      if (!this.leakyReconnectTimer) {
+        this.leakyReconnectTimer = setInterval(() => {
```

Comment:
Seems to me we should probably call `stopInterval()` at some point so it's not just sitting there counting in the background when the `hardReconnectCount` is 0

Reply:
done